var/spack/repos/builtin/packages/py-statsmodels/package.py

```diff
@@ -12,19 +12,26 @@ class PyStatsmodels(PythonPackage):
     homepage = "http://www.statsmodels.org"
     url      = "https://pypi.io/packages/source/s/statsmodels/statsmodels-0.8.0.tar.gz"
 
+    version('0.10.1', sha256='320659a80f916c2edf9dfbe83512d9004bb562b72eedb7d9374562038697fa10')
     version('0.8.0', sha256='26431ab706fbae896db7870a0892743bfbb9f5c83231644692166a31d2d86048')
 
     variant('plotting', default=False, description='With matplotlib')
 
+    depends_on('python@:3.6', when='@:0.8.0', type=('build', 'run'))
+
     # according to http://www.statsmodels.org/dev/install.html earlier versions
     # might work.
     depends_on('py-setuptools@0.6c5:', type='build')
-    depends_on('py-numpy@1.7.0:',      type=('build', 'run'))
-    depends_on('py-scipy@0.11:',       type=('build', 'run'))
-    depends_on('py-pandas@0.12:',      type=('build', 'run'))
-    depends_on('py-patsy@0.2.1:',      type=('build', 'run'))
-    depends_on('py-cython@0.24:',      type=('build', 'run'))
-    depends_on('py-matplotlib@1.3:',   type='run', when='+plotting')
+    depends_on('py-matplotlib@1.3:',   type=('build', 'run'), when='@0.8.0 +plotting')
+    depends_on('py-matplotlib@2.2:',   type=('build', 'run'), when='@0.10.1 +plotting')
+    depends_on('py-numpy@1.11.0:',     type=('build', 'run'), when='@0.10.1')
+    depends_on('py-numpy@1.7.0:',      type=('build', 'run'), when='@0.8.0')
+    depends_on('py-pandas@0.12:',      type=('build', 'run'), when='@0.8.0')
+    depends_on('py-pandas@0.19:',      type=('build', 'run'), when='@0.10.1')
+    depends_on('py-patsy@0.2.1:',      type=('build', 'run'), when='@0.8.0')
+    depends_on('py-patsy@0.4.0:',      type=('build', 'run'), when='@0.10.1')
+    depends_on('py-scipy@0.11:',       type=('build', 'run'), when='@0.8.0')
+    depends_on('py-scipy@0.18:',       type=('build', 'run'), when='@0.10.1')
 
     # TODO: Add a 'test' deptype
     # depends_on('py-nose', type='test')
```

Comment:
There is actually a `type='test'` now if you want to uncomment this.

Reply:
Truth be told, I'd rather leave it commented out....

That's code for "Now that I try it, it doesn't pass it's test.  Sigh...".  The errors happening over in scipy.  That code's changed on their tip of the tree, but if one steps back to their 1.3.x maint branch it lines up.

See https://github.com/statsmodels/statsmodels/issues/6263